exercises/rect/rect_test.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestRect(t *testing.T) {
+	for _, tc := range []struct {
+		x, y int
+		want [][]int
+	}{
+		{0, 0, [][]int{}},
```

Comment:
There should be more trivial tests, e.g. where both dimensions are `1`, one of the dimensions is `1` or `0`.